api/src/prescription/target-profile/infrastructure/repositories/target-profile-administration-repository.js

```diff
@@ -56,4 +56,33 @@ const getTubesByTargetProfileId = async (targetProfileId) => {
   return knexConn('target-profile_tubes').select('tubeId', 'level').where('targetProfileId', targetProfileId);
 };
 
-export { create, getTubesByTargetProfileId, update };
+const findByOrganization = async function ({ organizationId }) {
+  const results = await knex('target-profiles')
+    .select({
+      id: 'target-profiles.id',
+      name: 'target-profiles.name',
+      outdated: 'target-profiles.outdated',
+      isPublic: 'target-profiles.isPublic',
+      ownerOrganizationId: 'target-profiles.ownerOrganizationId',
+      sharedOrganizationId: 'target-profile-shares.organizationId',
+    })
+    .leftJoin('target-profile-shares', function () {
+      this.on('target-profile-shares.targetProfileId', 'target-profiles.id').on(
+        'target-profile-shares.organizationId',
+        organizationId,
+      );
+    })
+    .where({ outdated: false })
+    .where((qb) => {
+      qb.orWhere({ isPublic: true });
+      qb.orWhere({ ownerOrganizationId: organizationId });
+      qb.orWhere((subQb) => {
+        subQb.whereNotNull('target-profile-shares.id');
+      });
+    })
+    .orderBy('id', 'ASC');
+
+  return results.map((attributes) => new TargetProfileSummaryForAdmin(attributes));
```

Comment:
**Suggestion :** Si cette méthode repo utilise le model `TargetProfileSummaryForAdmin`, je suggère de la laisser dans `target-profile-summary-for-admin-repository.js`

Reply:
On a déja un repo `target-profile-administration-repository` qui peut accueillir cette méthode, plutôt que de créer un deuxième repo avec juste une méthode (et qu'on voudra probablement fusionner après)
Je propose de laisser comme ça et de renommer le repo plus tard.